### PR TITLE
Pipeline uppercase & lowercase

### PR DESCRIPTION
**Pull request by Exclamation-mark**
Pipeline uppercase modifier & lowercase modifier implementation  (#2)

I tried adding unit tests, which will benefit TDD developers. like this: 

```rust

#[cfg(test)]
mod tests {
    use super::*;

    #[tokio::test]
    async fn lowercase_works() {
        let ctx = Ctx::initial_state_with_value(Value::String(String::from("AbcD")));
        assert_eq!(
            LowerCaseItem::new()
                .call(ctx.clone())
                .await
                .unwrap()
                .value
                .as_str()
                .unwrap(),
            "abcd");
    }

    #[tokio::test]
    async fn should_check_ctx_value() {
        let ctx = Ctx::initial_state_with_value(Value::Null);
        let r = LowerCaseItem::new().call(ctx.clone()).await;
        assert!(r.is_err());
    }
}
```

### CODE REVIEW

**Comment by victorteokw** [comment]
Hi @Exclamation-mark let's call this `toLowerCase` and `toUpperCase`. Add a trailing new line to each end of files that you've added. The tests look all good. 👏

**Comment by victorteokw** [comment]
@luanxuechao review 👍

File: src/parser/std/pipeline/string/transform.rs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 use std::sync::Arc;
 use crate::core::pipeline::item::Item;
 use crate::core::pipeline::items::string::transform::ellipsis::EllipsisItem;
+use crate::core::pipeline::items::string::transform::uppercase::UpperCaseItem;
```
Comment (victorteokw) [inline src/parser/std/pipeline/string/transform.rs:4]: Rename this file to `to_upper_case`, rename this item to `ToUpperCaseItem`

Comment (Exclamation-mark) [reply src/parser/std/pipeline/string/transform.rs]: Sorry I just saw it now, I've fixed it.

File: src/core/pipeline/items/string/transform/to_upper_case.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait]
+impl Item for UpperCaseItem {
+    async fn call<'a>(&self, ctx: Ctx<'a>) -> Result<Ctx<'a>> {
+        match ctx.get_value() {
+            Value::String(ref s) =>
```
Comment (victorteokw) [inline src/core/pipeline/items/string/transform/to_upper_case.rs:19]: replace ref s to s

File: src/core/pipeline/items/string/transform/to_upper_case.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                Ok(ctx.with_value(Value::String(s.to_uppercase().to_owned()))),
```
Comment (victorteokw) [inline src/core/pipeline/items/string/transform/to_upper_case.rs:20]: Remove to owned

File: src/parser/std/pipeline/string/transform.rs [hunk 4]
```diff
@@ -48,6 +50,14 @@ pub(crate) fn to_word_case(_args: Vec<Argument>) -> Arc<dyn Item> {
     Arc::new(ToWordCaseItem::new())
 }
 
+pub(crate) fn to_lower_case(args: Vec<Argument>) -> Arc<dyn Item> {
+    Arc::new(LowerCaseItem::new())
```
Comment (victorteokw) [inline src/parser/std/pipeline/string/transform.rs:54]: Rename like this ToUpperCaseItem

**Pull request merged**